louve_addons/edi_purchase_diapar/views/res_partner_view.xml

```diff
@@ -15,5 +15,17 @@
             </field>
         </record>
 
+        <record id="invoice_supplier_form_inherit_diapar" model="ir.ui.view">
+            <field name="name">invoice.supplier.form.inherit.diapar</field>
+            <field name="model">account.invoice</field>
+            <field name="inherit_id" ref="account.invoice_supplier_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="//header/button[@name='action_cancel_draft']" position="after">
+                    <field name="partner_is_edi" invisible="1" />
+                    <button name="%(invoice_supplier_price_update_act)d" type="action" string="Update Prices from EDI" class="btn btn-primary" attrs="{'invisible':[('partner_is_edi','=', False)]}"/>
```

Comment:
We only should allow this button on draft invoices.
It's very important. We don't want this action to run on validated invoices.

Please also add a check in the function, and return an exception if the invoice is not in draft